pymc/sampling.py

```diff
@@ -655,31 +656,41 @@ def sample(
         return trace
 
 
-def _check_start_shape(model, start):
-    if not isinstance(start, dict):
-        raise TypeError("start argument must be a dict or an array-like of dicts")
-
-    # Filter "non-input" variables
-    initial_point = model.initial_point
-    start = {k: v for k, v in start.items() if k in initial_point}
+def _check_start_shape(model, start: PointType):
+    """Checks that the prior evaluations and initial points have identical shapes.
 
+    Parameters
+    ----------
+    model : pm.Model
+        The current model on context.
+    start : dict
+        The complete dictionary mapping (transformed) variable names to numeric initial values.
+    """
     e = ""
     for var in model.basic_RVs:
-        var_shape = model.fastfn(var.shape)(start)
-        if var.name in start.keys():
-            start_var_shape = np.shape(start[var.name])
-            if start_var_shape:
-                if not np.array_equal(var_shape, start_var_shape):
-                    e += "\nExpected shape {} for var '{}', got: {}".format(
-                        tuple(var_shape), var.name, start_var_shape
-                    )
-            # if start var has no shape
+        try:
+            var_shape = model.fastfn(var.shape)(start)
+            if var.name in start.keys():
+                start_var_shape = np.shape(start[var.name])
+                if start_var_shape:
+                    if not np.array_equal(var_shape, start_var_shape):
+                        e += "\nExpected shape {} for var '{}', got: {}".format(
+                            tuple(var_shape), var.name, start_var_shape
+                        )
+                # if start var has no shape
+                else:
+                    # if model var has a specified shape
+                    if var_shape.size > 0:
+                        e += "\nExpected shape {} for var " "'{}', got scalar {}".format(
+                            tuple(var_shape), var.name, start[var.name]
+                        )
+        except NotImplementedError as ex:
+            if ex.args[0].startswith("Cannot sample"):
+                _log.warning(
+                    "Unable to check start shape of {var} because the RV does not implement random sampling."
```

Comment:
Unrelated to this PR, but in theory we should be able to infer shapes without sampling, even for variables that cannot be sampled from...

Reply:
Totally. I opened a ticket #5031

Reply:
```suggestion
                   f"Unable to check start shape of {var} because the RV does not implement random sampling."
```